Enable building calendar for a particular seder

kalmanh opened the pull request:
Thought this would be neat. Let me know what you think.

תפסת מרובה לא תפסת. I am one of those guys that likes setting small goals. :)

For now, this will only work for seder nashim - it won't take me long to add others, if you think there is some merit to this PR.

kalmanh committed:
Enable building calendar for a particular seder

dbankier commented on index.js:
```diff
@@ -16,6 +16,11 @@ var total_m = _.reduce(flatArray, function(memo, num){ return memo + parseInt(nu
 
+// Sedarim
+var seder = {};
+seder.nashim = ["Yevamot", "Ketubot", "Nedarim", "Nazir", "Sotah", "Gittin", "Kidushin"];
```
The sedarim could/should be managed by the config files in `/data`.

dbankier commented on index.js:
```diff
@@ -85,13 +90,36 @@ exports.getToday = function (_date, pretty) {
+      return _.some(dailyQuota, partOfSeder);
+    }).map(function(dailyQuota){
+      return _.filter(dailyQuota, partOfSeder);
+    }).value();
```
A simple `reduce` function could do this instead of using underscore.

dbankier commented on index.js:
```diff
@@ -85,13 +90,36 @@ exports.getToday = function (_date, pretty) {
   var list = buildList(o.per_day || 2, o.perakim);
+
+  if(o.seder){
+    list = constrainListToSeder(list, o.seder);
```
Also you might want to do the limiting in the buildList function (more efficient).

kalmanh committed:
Move sedarim definition to data dir; move seder code to buildList to improve efficiency

kalmanh commented:
@dbankier - I addressed your concerns in the new commit.

Not sure what you meant by 

> A simple reduce function could do this instead of using underscore.

although I guess that's [OBE](http://www.acronymfinder.com/Overcome-By-Events-%28OBE%29.html) now.

If you can point me to what this "reduce function" thing is - I would appreciate it. I am more of a Java/Ruby/Rails guy - slowly getting over my aversion to JS with Node.js :)

dbankier commented:
Thank you
